### ISSUE TITLE
CallsTo A<object> does not consider any call to be valid :(

### ISSUE THREAD

**Issue by linvi** [FakeItEasy/FakeItEasy#205]
I have a method Func that takes a generic parameter T. As a result I did not want to pass the type itself, instead I wanted the callsto to expect an object of any type

As a result I tried : `.CallsTo(x => x.Func(A<object>.Ignored))`
The fake did not consider any call to be valid.

When I changed the method to be : `.CallsTo(x => x.Func(A<T>.Ignored))`
My fake worked correctly and was called.

In the meantime, I would still be quite happy to make the Fake.CallsTo to expect an object and not a generic type.

**Comment by blairconrad** [comment]
@adamralph, I agree with @patrik-hagne's comment that the call and method should not agree, although I still support returning false instead of throwing an exception.
I think that's the best takeaway from this: make sure consider the call not to have happened. Oh, and ideally the output of the failing `MustHaveHappened` should include enough information to show the two different signatures.

**Comment by blairconrad** [comment]
I apologize for the delay. I had forgotten about this issue. Testing with modern FakeItEasy (2.0.0-beta009), I found that the first test in https://github.com/FakeItEasy/FakeItEasy/issues/205#issuecomment-29887787 **passes**, once I rewrote it to compile:

```c#
[Test]
public void Problem1()
{
    var testable = A.Fake<ITestable>();

    A.CallTo(() => testable.GetObjectResult(A<ITestableObject>.Ignored))
        .Returns("success!");

    var result = testable.GetObjectResult(new object());

    Assert.That(result, Is.Not.EqualTo("success"));
    A.CallTo(() => testable.GetObjectResult(A<ITestableObject>.Ignored)).MustNotHaveHappened();
}
```

The second test continues to fail:

```c#
[Test]
public void Problem2()
{
    var testable = A.Fake<ITestable>();
    var testableObject = A.Fake<ITestableObject>();
    var testObject = new object();

    A.CallTo(() => testable.GetTypedResult(A<object>.Ignored))
        .ReturnsLazily(() =>
        {
            Console.WriteLine("hello");
            return "success!";
        });

    testable.GetTypedResult(testObject);
    testable.GetTypedResult(testableObject);

    A.CallTo(() => testable.GetTypedResult(A<object>.Ignored))
        .MustHaveHappened(Repeated.Exactly.Twice);
    // Method is invoked only once, making the test fail
}
```

with result 

```
FakeItEasy.ExpectationException : 

  Assertion failed for the following call:
    FakeItEasyQuestions.Issue203+ITestable.GetTypedResult<System.Object>(<Ignored>)
  Expected to find it exactly twice but found it #1 times among the calls:
    1: FakeItEasyQuestions.Issue203+ITestable.GetTypedResult<System.Object>(i: System.Object)
    2: FakeItEasyQuestions.Issue203+ITestable.GetTypedResult<FakeItEasyQuestions.Issue203+ITestableObject>(i: Faked FakeItEasyQuestions.Issue203+ITestableObject)
```

As @patrik-hagne pointed out in https://github.com/FakeItEasy/FakeItEasy/issues/205#issuecomment-30116571, the `GetTypedResult` is treated as two different methods, and this is arguably not a bug.

As @philippdolder suggested, we could explore adding open generic versions to some of the FakeItEasy methods to support this, although as @patrik-hagne, the existing `Where` can probably do just as well, and even @linvi seemed to agree, if I understand 

> But I believe this could be possible by using reflection. Checking the methods using the same name as well as the same number of parameters

correctly, from https://github.com/FakeItEasy/FakeItEasy/issues/205#issuecomment-30232397

I've run with this and the following test passes:

```c#

**Comment by adamralph** [comment]
We could look into it more but, IMHO, post 2.0.

**Comment by adamralph** [comment]
@linvi thanks for starting this discussion. We're now closing this in favour of #743 which represents the implementation of a way to specify an open generic call.
